src/cryptsetup/cryptsetup.c

```diff
@@ -131,13 +131,22 @@ static int parse_one_option(const char *option) {
                 }
 
         } else if ((val = startswith(option, "keyfile-offset="))) {
+                uint64_t off;
 
-                r = safe_atou(val, &arg_keyfile_offset);
+                r = safe_atou64(val, &off);
                 if (r < 0) {
                         log_error_errno(r, "Failed to parse %s, ignoring: %m", option);
                         return 0;
                 }
 
+                if ((size_t) off != off) {
```

Comment:
Is it not necessary to re-cast the value like `(uint64_t) (size_t) off != off`?

Reply:
ints are promoted to the longer type, so it's fine as is.

Reply:
Thanks.